rotkehlchen/tests/unit/decoders/test_metamask.py

```diff
@@ -0,0 +1,727 @@
+    fee_amount = '5.019297'
+    expected_events = [
+        EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal(gas_fees)),
+            location_label=user_address,
+            notes=f'Burned {gas_fees} ETH for gas',
+            counterparty=CPT_GAS,
+        ),
+        EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=1,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.TRADE,
+            event_subtype=HistoryEventSubType.SPEND,
+            asset=A_USDT,
+            balance=Balance(amount=FVal(swap_amount)),
+            location_label=user_address,
+            notes=f'Swap {swap_amount} USDT in {CPT_METAMASK} from {user_address}',
+            counterparty=CPT_METAMASK,
+            address=metamask_router_address,
+        ),
+        EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=2,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.TRADE,
+            event_subtype=HistoryEventSubType.RECEIVE,
+            asset=A_INU,
+            balance=Balance(amount=FVal(received_amount)),
+            location_label=user_address,
+            notes=f'Receive {received_amount} INU from {CPT_METAMASK} swap in {user_address}',
+            counterparty=CPT_METAMASK,
+            address=metamask_router_address,
+        ),
+        EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=141,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_USDT,
+            balance=Balance(amount=FVal(fee_amount)),
+            location_label=user_address,
+            notes=f'Deduct {fee_amount} USDT from {user_address} as {CPT_METAMASK} fees',
+            counterparty=CPT_METAMASK,
+            address=metamask_router_address,
+        ),
+    ]
+    assert expected_events == events
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('ethereum_accounts', [['0xDe0A989c715C594Eadc98a1b97a9aa65d3cECb48']])
+def test_metamask_swap_token_to_usdc(database, ethereum_inquirer, ethereum_accounts):
+    """Data taken from
+    https://etherscan.io/tx/0x17e8c0e123081c77b5bb553bd3cf553d4031547f5d8884b31c8d74bb76057add
+    """
+    tx_hash = deserialize_evm_tx_hash(
+        '0x17e8c0e123081c77b5bb553bd3cf553d4031547f5d8884b31c8d74bb76057add',
+    )
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=ethereum_inquirer,
+        database=database,
+        tx_hash=tx_hash,
+    )
+    metamask_router_address = string_to_evm_address(METAMASK_ROUTER_ETH)
```

Comment:
this `string_to_evm_address` should not be needed. Make the constant be the correct type beforehand, here and eslewhere